test/core/end2end/tests/http2_stats.cc

```diff
@@ -0,0 +1,257 @@
+  }
+
+ private:
+  static grpc_transport_stream_stats transport_stream_stats_
+      ABSL_GUARDED_BY(g_mu);
+};
+
+grpc_transport_stream_stats FakeServerCallTracer::transport_stream_stats_;
+
+class FakeServerCallTracerFactory : public ServerCallTracerFactory {
+ public:
+  ServerCallTracer* CreateNewServerCallTracer(Arena* arena) override {
+    return arena->ManagedNew<FakeServerCallTracer>();
+  }
+};
+
+// This test verifies the HTTP2 stats on a stream
+CORE_END2END_TEST(Http2FullstackSingleHopTest, StreamStats) {
+  g_mu = new Mutex();
+  CoreConfiguration::RegisterBuilder([](CoreConfiguration::Builder* builder) {
+    builder->channel_init()->RegisterFilter(GRPC_CLIENT_CHANNEL,
+                                            &FakeClientFilter::kFilter);
+  });
+  ServerCallTracerFactory::RegisterGlobal(new FakeServerCallTracerFactory);
+
+  auto send_from_client = RandomSlice(10);
+  auto send_from_server = RandomSlice(20);
+  CoreEnd2endTest::IncomingStatusOnClient server_status;
+  CoreEnd2endTest::IncomingMetadata server_initial_metadata;
+  CoreEnd2endTest::IncomingMessage server_message;
+  CoreEnd2endTest::IncomingMessage client_message;
+  CoreEnd2endTest::IncomingCloseOnServer client_close;
+  {
+    auto c = NewClientCall("/foo").Timeout(Duration::Minutes(5)).Create();
+    c.NewBatch(1)
+        .SendInitialMetadata({})
+        .SendMessage(send_from_client.Ref())
+        .SendCloseFromClient()
+        .RecvInitialMetadata(server_initial_metadata)
+        .RecvMessage(server_message)
+        .RecvStatusOnClient(server_status);
+    auto s = RequestCall(101);
+    Expect(101, true);
+    Step(Duration::Minutes(1));
+    s.NewBatch(102).SendInitialMetadata({}).RecvMessage(client_message);
+    Expect(102, true);
+    Step(Duration::Minutes(1));
+    s.NewBatch(103)
+        .SendStatusFromServer(GRPC_STATUS_UNIMPLEMENTED, "xyz", {})
+        .SendMessage(send_from_server.Ref())
+        .RecvCloseOnServer(client_close);
+    Expect(103, true);
+    Expect(1, true);
+    Step(Duration::Minutes(1));
+    EXPECT_EQ(s.method(), "/foo");
+  }
+  EXPECT_EQ(server_status.status(), GRPC_STATUS_UNIMPLEMENTED);
+  EXPECT_EQ(server_status.message(), "xyz");
+  EXPECT_FALSE(client_close.was_cancelled());
+  EXPECT_EQ(client_message.payload(), send_from_client);
+  EXPECT_EQ(server_message.payload(), send_from_server);
+
+  auto client_transport_stats =
+      FakeCallTracer::FakeCallAttemptTracer::transport_stream_stats();
+  EXPECT_EQ(client_transport_stats.outgoing.data_bytes,
+            send_from_client.size());
+  EXPECT_EQ(client_transport_stats.incoming.data_bytes,
+            send_from_server.size());
+  auto server_transport_stats = FakeServerCallTracer::transport_stream_stats();
+  EXPECT_EQ(server_transport_stats.outgoing.data_bytes,
+            send_from_server.size());
+  EXPECT_EQ(server_transport_stats.incoming.data_bytes,
+            send_from_client.size());
+  // TODO(yashykt): Add tests for framing bytes as well
```

Comment:
maybe let's just check that they're at least 23 on the client side? (9 bytes headers frame, 9 bytes data, 5 bytes grpc)?

Reply:
there is some stuff that needs to be fixed there, i'll prefer doing that in a follow-up PR